fe/fe-core/src/main/java/com/starrocks/sql/optimizer/rule/transformation/materialization/AggregatedMaterializedViewPushDownRewriter.java

```diff
@@ -396,6 +396,10 @@ public AggRewriteInfo visitLogicalAggregate(OptExpression optExpression, AggRewr
                     List<ScalarOperator> newArgs = aggCall.getChildren();
                     newArgs.set(0, newArg0);
                     String rollupFuncName = getRollupFunctionName(aggCall, false);
+                    if (rollupFuncName == null) {
```

Comment:
When will rollupFuncName be null?

Reply:
It happens when rollup+count(distinct)

```sql
CREATE TABLE test_pt (
  id int(11) NULL COMMENT "id",
  pt date NOT NULL COMMENT "",
  gmv int(11) NULL COMMENT "gmv"
) ENGINE=OLAP
DUPLICATE KEY(id)
COMMENT "OLAP"
PARTITION BY date_trunc('day', pt)
DISTRIBUTED BY HASH(pt)
PROPERTIES (
"replication_num" = "1"
);


CREATE MATERIALIZED VIEW test_pt_mv3
PARTITION BY (pt)
DISTRIBUTED BY hash(id) buckets 1
REFRESH ASYNC START("2024-03-08 03:00:00") EVERY(INTERVAL 1 DAY)
PROPERTIES (
"replication_num" = "1"
)
AS SELECT pt, id, gmv,count(1) AS pv
FROM test.test_pt
GROUP BY pt, id,gmv;


explain
select 
    count(distinct gmv) from test_pt
    where pt='2023-03-07' 
    and id IN (
        SELECT
            id
        FROM
            test_alias
    )
    group by
    id;

```

`count(distinct gmv)` will trigger this NPE when mv's group by keys contains `gmv` and the query's group by key is only `id` which is `rollup`

Reply:
Thank you for the explanation, better to add comment in the code base

Reply:
Added.